sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/csv/UnivocityParser.scala

```diff
@@ -317,7 +317,15 @@ class UnivocityParser(
         if (skipRow) {
           row.setNullAt(i)
         } else {
-          row(i) = valueConverters(i).apply(getToken(tokens, i))
+          // This is required to not set value as null ,
+          // 1. If the missing value at the end of line.
+          // 2. If the missing value at the beginning of line.
+          if (!options.naFilter && (i>=tokens.length ||
+            (i==0 && getToken(tokens, i).length == 0))) {
```

Comment:
```suggestion
            (i == 0 && getToken(tokens, i).length == 0))) {
```
should add a space between operator per the style guide